scripts/vscode.patch

```diff
@@ -599,6 +599,22 @@ index e22cfe1b45e..b48c1babf86 100644
  export class DiskFileSystemProvider extends AbstractDiskFileSystemProvider implements
  	IFileSystemProviderWithFileReadWriteCapability,
  	IFileSystemProviderWithOpenReadWriteCloseCapability,
+diff --git a/src/vs/platform/instantiation/common/instantiationService.ts b/src/vs/platform/instantiation/common/instantiationService.ts
+index f0db479e3c1..b8d93aae3f2 100644
+--- a/src/vs/platform/instantiation/common/instantiationService.ts
++++ b/src/vs/platform/instantiation/common/instantiationService.ts
+@@ -195,8 +195,10 @@ export class InstantiationService implements IInstantiationService {
+ 
+ 				if (instanceOrDesc instanceof SyncDescriptor) {
+ 					const d = { id: dependency.id, desc: instanceOrDesc, _trace: item._trace.branch(dependency.id, true) };
++					if (!graph.lookup(d)) {
```

Comment:
So if the service is already in the graph, we don't add it to the stack?

Reply:
It's interesting, because it would still "evaluate" the dependency, but not initialize it. It looks good to me.

Reply:
I'm a little confused by that code, because even if I remove the adding in the stack, everything works fine :shrug:

So I tried the safest change I can imagine, we'll see the reaction of the Microsoft team on it

> So if the service is already in the graph, we don't add it to the stack?

The service being on the graph means we already compute its dependencies at that point, right?